pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -49,10 +50,22 @@ func (n *NetworkPolicyController) updateClusterGroup(oldObj, curObj interface{})
 	klog.V(2).Infof("Processing UPDATE event for ClusterGroup %s", cg.Name)
 	newGroup := n.processClusterGroup(cg)
 	oldGroup := n.processClusterGroup(og)
+
+	selectorUpdated := getNormalizedNameForSelector(newGroup.Selector) != getNormalizedNameForSelector(oldGroup.Selector)
 	ipBlockUpdated := newGroup.IPBlock != oldGroup.IPBlock
 	svcRefUpdated := newGroup.ServiceReference != oldGroup.ServiceReference
-	if getNormalizedNameForSelector(newGroup.Selector) == getNormalizedNameForSelector(oldGroup.Selector) && !ipBlockUpdated && !svcRefUpdated {
-		// No change in the selectors of the ClusterGroup. No need to enqueue for further sync.
+	childGroupsUpdated := func(oldGroup, newGroup *antreatypes.Group) bool {
+		oldChildGroups, newChildGroups := sets.String{}, sets.String{}
+		for _, c := range oldGroup.ChildGroups {
+			oldChildGroups.Insert(c)
+		}
+		for _, c := range newGroup.ChildGroups {
+			newChildGroups.Insert(c)
+		}
+		return !oldChildGroups.Equal(newChildGroups)
+	}
+	if !selectorUpdated && !ipBlockUpdated && !svcRefUpdated && !childGroupsUpdated(oldGroup, newGroup) {
```

Comment:
Do you think we can do the same for other conditions, like
```
if newGroup.IPBlock == oldGroup.IPBlock  &&
  newGroup.ServiceReference == oldGroup.ServiceReference &&
  getNormalizedNameForSelector(newGroup.Selector) == getNormalizedNameForSelector(oldGroup.Selector) &&
  !childGroupsUpdated(oldGroup, newGroup)
```

I saw your comments that these should not be expensive, but considering we might need to process a large number of groups, I still think the optimization deserves the extra code complexity.

Reply:
Thanks for the suggestion @jianjuns. After a closer look, I realized that the `==` checks we had in place was actually wrong: the address of the ipBlocks cannot be identical. Fixed this logic and changed the code as you suggested